packages/terra-toggle-section-header/src/ToggleSectionHeader.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import SectionHeader from 'terra-section-header';
+import Toggle from 'terra-toggle';
+
+const VALID_LEVELS = [1, 2, 3, 4, 5, 6];
+
+const propTypes = {
+  /**
+   * Content in the body of the toggle section header component that will be expanded or hidden.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Text to be displayed in the section header.
+   */
+  title: PropTypes.string.isRequired,
+  /**
+   * Sets the heading level. One of 1, 2, 3, 4, 5, or 6.
+   */
+  level: PropTypes.oneOf(VALID_LEVELS),
+  /**
+   * Callback function triggered when toggle-section-header is closed.
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function triggered when toggle-section-header is opened.
+   */
+  onOpen: PropTypes.func,
+  /**
+   * Used to set props and HTML attributes on the toggle-section-header section-header.
+   */
+  selectionHeaderAttrs: PropTypes.object,
+  /**
+   * Sets the toggle-section-header to be animated when it is opened or closed.
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the toggle-section-header initial state to open.
+   */
+  isInitiallyOpen: PropTypes.bool,
+};
+
+const defaultProps = {
+  level: 2,
+  selectionHeaderAttrs: {},
+};
+
+class ToggleSectionHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { isInitiallyOpen = false } = this.props;
+    this.state = { isOpen: isInitiallyOpen };
+
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isInitiallyOpen !== this.props.isInitiallyOpen) {
+      this.setState({ isOpen: this.props.isInitiallyOpen });
```

Comment:
Is this line accurate? Shouldn't this be nextProps?

Also, do we need the if statement. Since isInitiallyOpen is a boolean, there are only two values, and I think setState on the same value won't do anything.

Reply:
I corrected the assignment here: 3b7c61ee5d4c24872c65cafb1717e44f62cb00ad

Though, I do see value in using the if-statement. Because of `setState`'s async nature, I think the call has the potential to still be accumulated and run in a batch of other `setState` calls, so even if the diff of the virtual-dom will end up doing nothing, I feel that the execution of the if-statement would be faster? I could be wrong though